### PR TITLE
TST:add test for df replace GH34871

### PR DESCRIPTION
**Pull request by nwweber**
- [x] improves upon #34871
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry <---- is that helpful for this specific regression test? happy to write this if you think this helps! :)

add tiny regression test for not failing on calling df.replace() with Period column

working on this revealed other quirks of df.replace(): it now does not raise an exception anymore, and the dataframe returned by replace() has the right values. However, the dtype of a Period column changes to Object, which is surprising. 

I've added the observations to GH34871

### CODE REVIEW

File: pandas/tests/frame/methods/test_replace.py [hunk 1]
```diff
@@ -1403,3 +1403,28 @@ def test_replace_with_duplicate_columns(self, replacement):
         result["B"] = result["B"].replace(7, replacement)
 
         tm.assert_frame_equal(result, expected)
+
+    def test_replace_period_ignore_float(self):
+        """
+        Regression test for GH#34871: if df.replace(1.0, 0.0) is called on a df
+        with a Period column the old, faulty behavior is to raise TypeError
```
Comment (jreback) [inline pandas/tests/frame/methods/test_replace.py:1410]: you don't need to reference older versions like this, this commit would only be on master.

File: pandas/tests/frame/methods/test_replace.py [hunk 2]
```diff
@@ -1403,3 +1403,28 @@ def test_replace_with_duplicate_columns(self, replacement):
...
+        (reported for Pandas 1.0.4). The intended behavior is to just ignore
+        that column. This has been fixed in newer versions. This is a regression
+        test that triggers for the old, broken behavior, i.e.
+        TypeError is raised
+        """
+        df = pd.DataFrame({"Per": [pd.Period("2020-01")] * 3})
```
Comment (jreback) [inline pandas/tests/frame/methods/test_replace.py:1416]: please use explicit comparisons, if this is *only* a test which is expected to fail then use ```@pytest.xfail(reason=....)```

Comment (jreback) [reply pandas/tests/frame/methods/test_replace.py:1416]: IOW don't comment out code

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/frame/methods/test_replace.py (hunk 2 above)
Comment (nwweber) [reply pandas/tests/frame/methods/test_replace.py:1416]: Hey! It's sort of the opposite, I expect being able to call df.replace(...) without it raising an exception for these specific inputs.

Would I then simply run "df.replace(...)" in the unit test, and that's sufficient?

Comment (jreback) [reply pandas/tests/frame/methods/test_replace.py:1416]: you would need to assert the results

Comment (nwweber) [reply pandas/tests/frame/methods/test_replace.py:1412]: updated, code now asserts expected result, also removed commented-out blocks of code

File: pandas/tests/frame/methods/test_replace.py (hunk 1 above)
Comment (nwweber) [reply pandas/tests/frame/methods/test_replace.py]: thanks for the advise, removed that part!

File: pandas/tests/frame/methods/test_replace.py [hunk 3]
```diff
@@ -1403,3 +1403,17 @@ def test_replace_with_duplicate_columns(self, replacement):
...
+
+    def test_replace_period_ignore_float(self):
+        """
+        Regression test for GH#34871: if df.replace(1.0, 0.0) is called on a df
+        with a Period column the old, faulty behavior is to raise TypeError.
+        """
+        df = pd.DataFrame({"Per": [pd.Period("2020-01")] * 3})
+        df_after_replace = df.replace(1.0, 0.0)
+
+        df_expected_result = pd.DataFrame({"Per": [pd.Period("2020-01")] * 3})
```
Comment (jreback) [inline pandas/tests/frame/methods/test_replace.py:1415]: can you call things result= and expected= its much easier to read (and doesn't mix in the same name)

File: pandas/tests/frame/methods/test_replace.py [hunk 4]
```diff
@@ -1403,3 +1403,17 @@ def test_replace_with_duplicate_columns(self, replacement):
...
+        # currently replace() changes dtype from Period to object
+        df_expected_result["Per"] = df_expected_result["Per"].astype(object)
```
Comment (jreback) [inline pandas/tests/frame/methods/test_replace.py:1417]: so this is wrong, we want to xfail this test, this *should* not change the dtype, pls add the issue number in the xfail itself

Comment (jreback) [reply pandas/tests/frame/methods/test_replace.py:1417]: meaning, yes it doesn't raise an expection, but the output is still converted to object

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/frame/methods/test_replace.py (hunk 3 above)
Comment (nwweber) [reply pandas/tests/frame/methods/test_replace.py]: done!

File: pandas/tests/frame/methods/test_replace.py (hunk 4 above)
Comment (nwweber) [reply pandas/tests/frame/methods/test_replace.py]: gotcha, thanks for teaching me about xfail. i changed the test to now expect the result we would like to see from replace() in the future, and to xfail for now.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @nwweber 

happy for you to tackle https://github.com/pandas-dev/pandas/issues/34871 :->

**Comment by nwweber** [comment]
> thanks @nwweber
> 
> happy for you to tackle #34871 :->

i'll have a look over the weekend :)

**Pull request merged**